ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/utils/KsqlPlanSchemaGenerator.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server.utils;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.google.common.collect.ImmutableMap;
+import com.kjetland.jackson.jsonSchema.JsonSchemaConfig;
+import com.kjetland.jackson.jsonSchema.JsonSchemaGenerator;
+import io.confluent.ksql.engine.KsqlPlan;
+import io.confluent.ksql.execution.expression.tree.Expression;
+import io.confluent.ksql.execution.expression.tree.FunctionCall;
+import io.confluent.ksql.execution.json.PlanJsonMapper;
+import io.confluent.ksql.execution.plan.ExecutionStep;
+import io.confluent.ksql.execution.plan.SelectExpression;
+import io.confluent.ksql.execution.windows.KsqlWindowExpression;
+import io.confluent.ksql.schema.ksql.ColumnRef;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+
+public final class KsqlPlanSchemaGenerator {
+  private static final ObjectMapper MAPPER = PlanJsonMapper.create();
+
+  private static final Map<Class<?>, JsonNode> POLYMORPHIC_TYPES = ImmutableMap.of(
+      ExecutionStep.class, definitionForPolymorphicType(ExecutionStep.class)
+  );
+
+  private static JsonSchemaConfig configure() {
+    final JsonSchemaConfig vanilla = JsonSchemaConfig.vanillaJsonSchemaDraft4();
+    return JsonSchemaConfig.create(
+        vanilla.autoGenerateTitleForProperties(),
+        Optional.empty(),
+        false,
+        false,
+        vanilla.usePropertyOrdering(),
+        vanilla.hidePolymorphismTypeProperty(),
+        vanilla.disableWarnings(),
+        vanilla.useMinLengthForNotNull(),
+        vanilla.useTypeIdForDefinitionName(),
+        Collections.emptyMap(),
+        vanilla.useMultipleEditorSelectViaProperty(),
+        Collections.emptySet(),
+        // the schema generator doesn't play nice with custom serializers, so we add a
+        // config to remap the custom-serialized types to their underlying primitive
+        new ImmutableMap.Builder<Class<?>, Class<?>>()
```

Comment:
Is this only a temporary thing?  What does this mean in terms of:

a) the schema generated: are we losing schema validation on these types?
b) the serialized form: how do these types look when serialized?

Reply:
> a) the schema generated: are we losing schema validation on these types?
Yes - from a jsonschema POV we can't do much better than defining these schemas as strings. I think we can add a `contentMediaType` that describes what the string contains (e.g. `text/sqlexpression`, `text/sqlschema`, etc). Let me give that a go.

> b) the serialized form: how do these types look when serialized?
These are all serialized as text parseable by the ksql parser.